File: java/src/main/java/com/example/java/bean/ShareData.java
package com.example.java.bean;

import org.litepal.annotation.Column;
import org.litepal.crud.LitePalSupport;

/**
 * @作者 陈忠岳
 * @主要功能
 * @创建日期 6/2/22
 */

public class ShareData extends LitePalSupport {
    public String getSymbol() {
        return symbol;
    }

    public void setSymbol(String symbol) {
        this.symbol = symbol;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Double getValue() {
        return value;
    }

    public void setValue(Double value) {
        this.value = value;
    }

    public Double getChange() {
        return change;
    }

    public void setChange(Double change) {
        this.change = change;
    }

    public Double getmFloat() {
        return mFloat;
    }

    public void setmFloat(Double mFloat) {
        this.mFloat = mFloat;
    }

    public String getDescribe() {
        return describe;
    }

    public void setDescribe(String describe) {
        this.describe = describe;
    }

    public Boolean getAdd() {
        return add;
    }

    public void setAdd(Boolean add) {
        this.add = add;
    }

    public Boolean getOptional() {
        return optional;
    }

    public void setOptional(Boolean optional) {
        this.optional = optional;
    }

    public int getType() {
        return type;
    }

    public void setType(int type) {
        this.type = type;
    }

    @Column(unique = true)
    private String symbol = "";//股票码
    private String name = "";//股票名
    private Double value = 0.0;//现价
    private Double change = 0.0;//增长值
    private Double mFloat = 0.0;//涨幅值
    private String describe = "";//额外说明
    private Boolean add = false;//收藏
    private Boolean optional = false;//是否关注
    private int type = 1;//A股 1 美股 3 港股 4
}

File: kotlin/src/main/java/com/example/kotlin/bean/ShareData.kt
package com.example.kotlin.bean

import com.google.gson.annotations.SerializedName
import org.litepal.annotation.Column
import org.litepal.crud.LitePalSupport


/**
 *行情请求 自定义转换
 *
 *
 * var hq_str_s_sh601988="中国银行,3.640,0.040,1.11,943238,34177";
 *                            股票码    股票名     现价 增长值 涨幅值
 *
 * 2,002477,雏鹰退,0.17,0.01,6.25,773719,13063057,18.75,0.18,0.15,0.16,0.16,0.00,1.67,3.91,-,-1.20,5329771
43,336699439,-80.68%,-88.96%,6.25,2010-09-15,2019-10-14 15:00:00,773719
 */
data class ShareData(
    @Column(unique = true)
    var symbol: String = "", //股票码
    var name: String = "",  //股票名
    var value: Double = 0.0, //现价
    var change: Double = 0.0, //增长值
    var float: Double = 0.0, //涨幅值
    var describe: String = "",//额外说明
    var add: Boolean = false,//收藏
    var optional: Boolean = false,//是否关注
    var type : Int = 1 //A股 1 美股 3 港股 4
): LitePalSupport(){}